packages/glibc.rb

```diff
@@ -3,169 +3,257 @@
-      when 'i686'
-        system "../configure",
-               "--prefix=#{CREW_PREFIX}",
-               "--libdir=#{CREW_LIB_PREFIX}",
-               "--with-headers=#{CREW_PREFIX}/include",
-               "--disable-werror",
-               "--disable-sanity-checks",
-               "--enable-shared",
-               "--disable-multilib",
-               "--enable-obsolete-rpc",
-               "libc_cv_forced_unwind=yes",
-               "libc_cv_ssp=no",
-               "libc_cv_ssp_strong=no"
+    system 'mkdir -pv glibc_build'
+    Dir.chdir 'glibc_build' do
+      case LIBC_VERSION
+      when '2.23'
+        case ARCH
+        when 'armv7l', 'aarch64'
+          system '../configure',
+                 "--prefix=#{CREW_PREFIX}",
+                 "--libdir=#{CREW_LIB_PREFIX}",
+                 "--with-headers=#{CREW_PREFIX}/include",
+                 '--disable-werror',
+                 '--disable-sanity-checks',
+                 '--enable-shared',
+                 'libc_cv_forced_unwind=yes',
+                 '--without-selinux'
+        when 'x86_64'
+          system "echo \"slibdir=#{CREW_LIB_PREFIX}\" > configparms"
+          puts "echo \"slibdir=#{CREW_LIB_PREFIX}\" > configparms"
+          system '../configure',
+                 "--prefix=#{CREW_PREFIX}",
+                 "--libdir=#{CREW_LIB_PREFIX}",
+                 "--with-headers=#{CREW_PREFIX}/include",
+                 '--disable-werror',
+                 '--disable-sanity-checks',
+                 '--enable-shared',
+                 '--disable-multilib',
+                 'libc_cv_forced_unwind=yes',
+                 'libc_cv_ssp=no',
+                 'libc_cv_ssp_strong=no'
+        when 'i686' # This is only for glibc 2.23
+          system '../configure',
+                 "--prefix=#{CREW_PREFIX}",
+                 "--libdir=#{CREW_LIB_PREFIX}",
+                 "--with-headers=#{CREW_PREFIX}/include",
+                 '--disable-werror',
+                 '--disable-sanity-checks',
+                 '--enable-shared',
+                 '--disable-multilib',
+                 'libc_cv_forced_unwind=yes',
+                 'libc_cv_ssp=no',
+                 'libc_cv_ssp_strong=no'
+        end
+      when '2.32'
+        # Optimization flags from https://github.com/InBetweenNames/gentooLTO
+        case ARCH
+        when 'armv7l', 'aarch64'
+          system "env CFLAGS='-pipe -O2 -fipa-pta -fno-semantic-interposition -fdevirtualize-at-ltrans' \
+            LD=ld ../configure \
+            --prefix=#{CREW_PREFIX} \
+            --libdir=#{CREW_LIB_PREFIX} \
+            --with-headers=#{CREW_PREFIX}/include \
+            --disable-profile \
+            --disable-sanity-checks \
+            --disable-werror \
+            --enable-bind-now \
+            --enable-kernel=3.18 \
+            --enable-shared \
+            --enable-static-pie \
+            libc_cv_c_cleanup=yes \
+            libc_cv_forced_unwind=yes \
+            "
```

Comment:
Does the arm build work?

Reply:
Rejiggering this to not build non-locale binaries in lib/... so maybe.

Reply:
Can't test the armv7l build I just made without a glibc 2.32 arch RootFS, but the x86_64 build works the same way, so it should in theory. Really this is mostly providing headers...